fdk.go

```diff
@@ -107,6 +111,84 @@ func doHTTP(handler Handler, ctx context.Context, in io.Reader, out io.Writer) {
 	}
 }
 
+func doJSON(handler Handler, ctx context.Context, in io.Reader, out io.Writer) {
+	var buf bytes.Buffer
+	hdr := make(http.Header)
+
+	for {
+		doJSONOnce(handler, ctx, in, out, &buf, hdr)
+	}
+}
+
+type jsonio struct {
+	Body        string `json:"body"`
+	ContentType string `json:"content_type"`
+}
+
+type callRequestHTTP struct {
+	Type       string      `json:"type"`
+	RequestURL string      `json:"request_url"`
+	Headers    http.Header `json:"headers"`
+}
+
+type jsonIn struct {
+	jsonio
+	CallID   string           `json:"call_id"`
+	Protocol *callRequestHTTP `json:"protocol"`
+}
+
+type callResponseHTTP struct {
+	StatusCode int         `json:"status_code,omitempty"`
+	Headers    http.Header `json:"headers,omitempty"`
+}
+
+type jsonOut struct {
+	jsonio
+	Protocol *callResponseHTTP `json:"protocol,omitempty"`
```

Comment:
this needn't be a pointer, we'll get better locality if it's a value and from what I can tell this is never nil (the json will always be displayed as an object just the same for a value as a non-nil value pointer)